scripts/ci/deploy

```diff
@@ -0,0 +1,93 @@
+#!/bin/bash
+
+set -o errexit
+
+declare ERR_MISOPT=1 \
+	ERR_AWSCLI=2 \
+	ERR_CONFIG=3 \
+	ERR_REGION=4 \
+	ERR_ENVIRON=5 \
+	ERR_UNKNOWN=255
+
+if ! $(which aws &>/dev/null); then
+	echo "error: aws is not found"
+	exit $ERR_AWSCLI
+fi
+
+declare timestamp=$(date "+%Y-%m-%dT%H:%M:%S") \
+	revision=$(git rev-parse --short HEAD)
+
+echo $revision >VERSION
+
+declare branch=${BRANCH:-$(git rev-parse --abbrev HEAD)}
+declare config=$(jq --raw-output ".${branch}.config" scripts/ci/map.json)
+
+if [[ -z "$config" || "$config" == "null" ]]; then
+	echo "error: config could not be set"
+	exit $ERR_CONFIG
+fi
+
+./configure --without-envvar
+source .env.sh
+make -C client dist
+
+git clean --force -d -x client/
+
+declare regions=$(jq --raw-output ".${branch}.environments | keys | .[]" scripts/ci/map.json)
+
+if [[ -z "$regions" ]]; then
+	echo "error: no regions found"
+	exit $ERR_REGION
+fi
+
+for region in $(echo $regions); do
+	declare environments=$(jq --raw-output ".${branch}.environments[\"${region}\"] | .[]" scripts/ci/map.json)
+
+	if [[ -z "$environments" ]]; then
+		echo "error: no environment is found"
+		exit $ERR_ENVIRON
+	fi
+
+	while read -r environment; do
+		declare archive_name=$timestamp-$revision
+		declare version_label=$archive_name-$branch-$environment
+		declare archive_file=$version_label.zip
+
+		./configure \
+			--without-envvars \
+			--config $config \
+			--ebEnvName $environment
+
+		source .env.sh
```

Comment:
This is a must to read credentials for environment specific credentials. L31 is probably not necessary, though. Good point.